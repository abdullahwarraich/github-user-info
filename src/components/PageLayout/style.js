import styled from 'styled-components'

export const ContentDiv = styled.div`
  height: 100%;
  display: flex;
  padding: 24px;
  background: #fff;
`