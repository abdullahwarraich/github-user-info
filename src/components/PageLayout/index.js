import React from 'react'
import { Layout } from 'antd'
import AppFooter from '../Footer'
import Navbar from '../Navbar'
import { ContentDiv } from './style'

const { Content } = Layout

const PageLayout = ({ children }) => (
  <Layout className='layout'>
    <Navbar />
    <Content>
      <ContentDiv>{children}</ContentDiv>
    </Content>
    <AppFooter />
  </Layout>
)

export default PageLayout
