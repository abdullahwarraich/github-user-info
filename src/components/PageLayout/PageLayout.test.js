import { render } from '@testing-library/react'
import PageLayout from './index'

describe('PageLayout', () => {
  let pageLayoutRender

  afterEach(() => {
    pageLayoutRender.unmount()
  })

  test('should render page layout component', () => {
    pageLayoutRender = render(
      <PageLayout>
        <h1>{'Header'}</h1>
        <p>{'Paragraph'}</p>
        <div>{'Div'}</div>
      </PageLayout>
    )
    expect(pageLayoutRender).toMatchSnapshot()
  })
})
