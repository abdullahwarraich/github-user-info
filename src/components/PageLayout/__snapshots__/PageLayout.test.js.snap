// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout should render page layout component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="ant-layout layout"
      >
        <div
          class="sc-eCImvq ghfczK"
        >
          <nav
            class="sc-bdvvaa iwfTtt"
          >
            <div
              class="sc-gsDJrp gPfrBN"
            >
              <div
                class="leftMenu"
              >
                <ul
                  class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
                  role="menu"
                >
                  <li
                    class="ant-menu-item ant-menu-item-only-child"
                    role="menuitem"
                    style="padding-left: 24px;"
                  >
                    Profile
                  </li>
                  <li
                    class="ant-menu-item ant-menu-item-only-child"
                    role="menuitem"
                    style="padding-left: 24px;"
                  >
                    Gists
                  </li>
                  <li
                    class="ant-menu-item ant-menu-item-only-child"
                    role="menuitem"
                    style="padding-left: 24px;"
                  >
                    Orgs
                  </li>
                  <li
                    class="ant-menu-item ant-menu-item-only-child"
                    role="menuitem"
                    style="padding-left: 24px;"
                  >
                    Search New User
                  </li>
                </ul>
              </div>
              <button
                class="ant-btn ant-btn-primary barsMenu"
                type="button"
              >
                <span
                  class="sc-hKwCoD jeKmDZ"
                />
              </button>
            </div>
          </nav>
        </div>
        <main
          class="ant-layout-content"
        >
          <div
            class="sc-jRQAMF eCZEON"
          >
            <h1>
              Header
            </h1>
            <p>
              Paragraph
            </p>
            <div>
              Div
            </div>
          </div>
        </main>
        <footer
          class="ant-layout-footer"
          style="text-align: center;"
        >
          Github User Info ©2021 Created by Abdullah Warraich
        </footer>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="ant-layout layout"
    >
      <div
        class="sc-eCImvq ghfczK"
      >
        <nav
          class="sc-bdvvaa iwfTtt"
        >
          <div
            class="sc-gsDJrp gPfrBN"
          >
            <div
              class="leftMenu"
            >
              <ul
                class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
                role="menu"
              >
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  role="menuitem"
                  style="padding-left: 24px;"
                >
                  Profile
                </li>
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  role="menuitem"
                  style="padding-left: 24px;"
                >
                  Gists
                </li>
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  role="menuitem"
                  style="padding-left: 24px;"
                >
                  Orgs
                </li>
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  role="menuitem"
                  style="padding-left: 24px;"
                >
                  Search New User
                </li>
              </ul>
            </div>
            <button
              class="ant-btn ant-btn-primary barsMenu"
              type="button"
            >
              <span
                class="sc-hKwCoD jeKmDZ"
              />
            </button>
          </div>
        </nav>
      </div>
      <main
        class="ant-layout-content"
      >
        <div
          class="sc-jRQAMF eCZEON"
        >
          <h1>
            Header
          </h1>
          <p>
            Paragraph
          </p>
          <div>
            Div
          </div>
        </div>
      </main>
      <footer
        class="ant-layout-footer"
        style="text-align: center;"
      >
        Github User Info ©2021 Created by Abdullah Warraich
      </footer>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
